portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java

```diff
@@ -0,0 +1,53 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.sps.data.Marker;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Handles fetching and saving markers data. */
+@WebServlet("/markers")
+public class MarkerServlet extends HttpServlet {
+
+  private List<Marker> markersOnMap = new ArrayList<>();
+
+  /** Responds with a JSON array containing marker data. */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+
+    Gson gson = new Gson();
+    String json = gson.toJson(markersOnMap);
+
+    response.getWriter().println(json);
```

Comment:
and then you can just condense this to response.getWriter().println(gso.toJson(markersOnMap));